python/tvm/auto_scheduler/task_scheduler.py

```diff
@@ -659,3 +660,87 @@ def post_tune(self, task_scheduler, task_id):
                 )
             )
             filep.flush()
+
+
+class VisualizeProgress(TaskSchedulerCallback):
+    """Callback that generates a plot to visualize the tuning progress.
+
+    Parameters
+    ----------
+    title: str
+        Specify the title of the matplotlib figure.
+    si_prefix: str
+        SI prefix for flops
+    keep_open: bool
+        Wait until the matplotlib window was closed by the user.
+    live: bool
+        If false, the graph is only written to the file specified in out_path.
+    out_path: str
+        Path where the graph image should be written (if defined).
+    """
+
+    def __init__(
+        self,
+        title="AutoScheduler Progress",
+        si_prefix="G",
+        keep_open=False,
+        live=True,
+        out_path=None,
+    ):
+        self.best_flops = {}
+        self.all_cts = {}
+        self.si_prefix = si_prefix
+        self.keep_open = keep_open
+        self.live = live
+        self.out_path = out_path
+        self.init_plot(title)
+
+    def __del__(self):
```

Comment:
we shouldn't do all this stuff in `__del__`. Tracebacks aren't reported and it can be unpredictable when this is invoked. Can you change this to use `__enter__` and `__exit__` so it can be used with `with` statement, and add an explicit part in the `tvmc` tuning infra where we expect this file written?